### PR TITLE
Changes bessd_test to ensure that fork() behavior is self-contained, …

### PR DESCRIPTION
**Pull request by barath**
…helping to ensure that tests that fail propagate that error to the top level.

### CODE REVIEW

File: core/bessd_test.cc [hunk 1]
```diff
@@ -116,11 +117,12 @@ TEST(TryAcquirePidfileLock, AlreadyHeld) {
     ASSERT_EQ(0, flock(fd, LOCK_EX | LOCK_NB)) << "Couldn't acquire file lock for test.";
 
     // Sleep for up to 10 seconds, while holding lock.
-    sleep(10);
+    sleep(3);
```
Comment (sangjinhan) [inline core/bessd_test.cc]: Would it be possible to remove this arbitrary (and potentially unreliable) delay, e.g., with some IPC? A few second delay makes me a little hesitant to run the ./all_test script frequently.

Comment (barath) [inline core/bessd_test.cc]: I was thinking about that.  Sure, I can see what will work best.

**Comment by barath** [comment]
I changed the behavior of bessd_test so it now uses pipes instead of sleeps to coordinate between child and parent processes in tests.  (Unfortunately DEATH tests didn't appear to work properly when invoked inside of lambdas, so I ended up not being able to use a C++11-style solution.)

**Comment by sangjinhan** [comment]
Merged, thanks!

**Pull request merged**